inst/rmd/lua/short-title.lua

```diff
@@ -0,0 +1,17 @@
+function Header(el)
+  local options = el.attributes['short-title']
+  if options == nil then return nil end
```

Comment:
We need to check the JSS requirements (https://www.jstatsoft.org/pages/view/authors) and see if plain short titles are always required. If so, we need to generate a plain-text version (no formatting commands) of the header and use the text as the short title.

Reply:
Only `\section` using special latex syntax use a short title like
```tex
\section[Using BibTeX]{Using \textsc{Bib}{\TeX}} \label{app:bibtex}
```
However, they often use `\label` (not mandatory too).  Pandoc adds a label for each header. We could also do that, but we need to generate an identifier from the text.

Reply:
`\label` has been added based ton the header identifier built by Pandoc itself.

Reply:
However, I don't really know if header identifier is always ok. In pandoc a special `toLabel` function is used to create this `\label{}` part, that calls `stringToLaTeX` that aims at "escape things as needed for LaTeX"
So I guess, the header identifier won't always be ok. 🤔 

As a fail safe, we could add a label attributes - if provided we use that as label ? Or maybe it is was the `{#id}` is for.

Reply:
> As a fail safe, we could add a label attributes - if provided we use that as label ? Or maybe it is was the {#id} is for. I'll check.

Yes - it works. 
```md
# \proglang{R} code {short-title="R code" #r-code}
```
will be correctly transformed to 
```tex
\section[R code]{\proglang{R} code}\label{r-code}
```